src/MiddlewareListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Replacement for the Silex MiddlewareListener to allow arbitrary injection into middleware functions.
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class MiddlewareListener extends \Silex\EventListener\MiddlewareListener
+{
+    /**
+     * @var BeforeMiddlewareCaller
+     */
+    private $beforeMiddlewareCaller;
+
+    /**
+     * @var AfterMiddlewareCaller
+     */
+    private $afterMiddlewareCaller;
+
+    /**
+     * @param Application            $app                    The application
+     * @param BeforeMiddlewareCaller $beforeMiddlewareCaller The caller that handles resolving and injecting before middlewares
+     * @param AfterMiddlewareCaller  $afterMiddlewareCaller  The caller that handles resolving and injecting after middlewares
+     */
+    public function __construct(Application $app, BeforeMiddlewareCaller $beforeMiddlewareCaller, AfterMiddlewareCaller $afterMiddlewareCaller)
+    {
+        parent::__construct($app);
+        $this->beforeMiddlewareCaller = $beforeMiddlewareCaller;
+        $this->afterMiddlewareCaller = $afterMiddlewareCaller;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $routeName = $request->attributes->get('_route');
+        if (!$route = $this->app['routes']->get($routeName)) {
+            return;
+        }
+
+        foreach ((array) $route->getOption('_before_middlewares') as $callback) {
+
+            $ret = $this->beforeMiddlewareCaller->call($callback, $request);
+
```

Comment:
Blank lines should be aligned with those of the `foreach` in `onKernelResponse()`.